forge-webhook-parser.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/python3
+"""
+Parse a webhook payload file to clone the git repository to
+the current directory.
+"""
+
+import argparse
+import json
+import shutil
+import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List, Optional
+
+
+@dataclass
+class PayloadFields:
+    """
+    Dataclass of details extracted from the payload.
+    """
+
+    added: List[str]
+    removed: List[str]
+    modified: List[str]
+    default_repo_url: Optional[str] = None
+    default_clone_path: Optional[Path] = None
+
+
+PAYLOAD_FILE = "./github-payload.json"
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-f", "--file", help="webhook payload file to parse", default=PAYLOAD_FILE
+)
+args = parser.parse_args()
+
+with open(args.file, "r", encoding="utf-8") as file:
+    data = json.load(file)
+
+repo_url_keys = []
+
+if "repository" in data and "clone_url" in data["repository"]:
+    repo_url_keys.append(data["repository"]["clone_url"])
+
+if "project" in data and "git_http_url" in data["project"]:
+    repo_url_keys.append(data["project"]["git_http_url"])
+
+default_repo_url = next((url for url in repo_url_keys if url), "")
+
+
+def extract_project_name() -> str:
+    """Extract the name of the project from the clone url."""
+    if default_repo_url:
+        remove_git = default_repo_url.rstrip(".git")
+        project_name = remove_git.split("/")[-1]
+        return project_name
+    return ""
+
+
+default_clone_path = Path.cwd() / extract_project_name()
+
+
+def clean_clone_path(clone_path: Path = default_clone_path):
+    """Remove already cloned repositories if they exist.
+
+    Args:
+        clone_path (Path): Directory of the cloned repo.
```

Comment:
I'd drop the data types from docstrings since you use static types. That way we won't have to worry that the docstring datatypes fall out of sync with reality.